SetupDataPkg/ConfApp/SetupConf.c

```diff
@@ -734,67 +697,23 @@ CreateXmlStringFromCurrentSettings (
     goto EXIT;
   }
 
-  DataSize = 0;
-  Status   = mSettingAccess->Get (
-                               mSettingAccess,
-                               DFCI_OEM_SETTING_ID__CONF,
-                               &mAuthToken,
-                               DFCI_SETTING_TYPE_BINARY,
-                               &DataSize,
-                               &Dummy,
-                               &Flags
-                               );
-  if (Status != EFI_BUFFER_TOO_SMALL) {
-    DEBUG ((DEBUG_ERROR, "Unexpected result from getting - %r\n", Status));
-    goto EXIT;
-  }
-
-  Data = AllocatePool (DataSize);
-  if (Data == NULL) {
-    Status = EFI_OUT_OF_RESOURCES;
-    goto EXIT;
-  }
-
-  Status = mSettingAccess->Get (
-                             mSettingAccess,
-                             DFCI_OEM_SETTING_ID__CONF,
-                             &mAuthToken,
-                             DFCI_SETTING_TYPE_BINARY,
-                             &DataSize,
-                             Data,
-                             &Flags
-                             );
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "Unexpected result from getting - %r\n", Status));
-    goto EXIT;
-  }
-
-  Status = IterateConfData (Data, CollectAllConfigTags);
+  Status = RetrieveActiveConfigVarList (&VarListEntries, &VarListEntriesCount);
   if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, "Iterate config data failed - %r\n", Status));
+    DEBUG ((DEBUG_ERROR, "Retrieving config data failed - %r\n", Status));
     goto EXIT;
   }
 
-  FreePool (Data);
-  Data = NULL;
-
   // Now get the individual settings
-  while (!IsListEmpty (&mCollectedConfigTags)) {
-    Link    = GetFirstNode (&mCollectedConfigTags);
-    TagNode = BASE_CR (Link, CONFIG_TAG_LINK_HEADER, Link);
-    TagId   = TagNode->TagId;
-    RemoveEntryList (Link);
-    FreePool (TagNode);
-
-    AsciiSize = sizeof (SINGLE_SETTING_PROVIDER_START) + 2 * sizeof (TagId);
+  for (Index = 0; Index < VarListEntriesCount; Index++) {
+    AsciiSize = StrnLenS (VarListEntries[Index].Name, DFCI_MAX_ID_LEN) + 1;

```

Comment:
Will this cause a CodeQL flag for not checking VarListEntries before accessing it? From the RetrieveActiveConfigVarList call, it will be okay, as the status would be set and VarListEntriesCount would be 0.

Reply:
Updated with more checks.